### PR TITLE
Dynamically generate conf.py with jupyter-book and remove from repo

### PR DESCRIPTION
**Pull request by benjaoming**
Re: https://github.com/readthedocs/readthedocs.org/issues/9405

I've tried to run this locally and it seems to work.

### CODE REVIEW

**Comment by pamoroso** [comment]
Thank you very much, I really appreciate it.

I applied the patch to a [test repo](https://github.com/pamoroso/jbrtd-test) that publishes the documentation to [Read The Docs](https://readthedocs.org/projects/jbrtd-test/) and has the same structure of my Suite8080 project. But on Read The Docs with it [fails](https://readthedocs.org/projects/jbrtd-test/builds/17397684/) to [build](https://readthedocs.org/projects/jbrtd-test/builds/17397684/) the same error.

**Comment by benjaoming** [comment]
Something is wrong with how you applied this change in your test repo. Your old conf.py is still there.

image

**Comment by pamoroso** [comment]
You're right, `conf.py` was a leftover file from a local build. Removing `conf.py` from the repo does make the Read The Docs build [complete successfully](https://readthedocs.org/projects/jbrtd-test/builds/17398104/). However, the [published documentation](https://jbrtd-test.readthedocs.io/en/latest/) is missing the content of my repo, displays a Sphinx project placeholder, and doesn't have the Jupyter Book style.

**Comment by benjaoming** [comment]
> However, the [published documentation](https://jbrtd-test.readthedocs.io/en/latest/) is missing the content of my repo, displays a Sphinx project placeholder, and doesn't have the Jupyter Book style.

At a quick glance, it looks fine? I don't see any placeholders?

**Comment by pamoroso** [comment]
This is what the [published documentation](https://jbrtd-test.readthedocs.io/en/latest/) looks like to me:

jbrtd-test

Instead, I would expect the home page to have [this text](https://github.com/pamoroso/jbrtd-test/blob/main/README.md) and [this formatting style](https://suite8080.readthedocs.io).

**Comment by benjaoming** [comment]
Let me get back to you, need to figure out why the `pre_build` step hasn't been run.

File: .readthedocs.yaml [hunk 1]
```diff
@@ -1,6 +1,18 @@
+# Set the version of Python and other tools you might need
+build:
```
Comment (humitos) [inline .readthedocs.yaml:2]:
```suggestion
version: 2

# Set the version of Python and other tools you might need
build:
```

**Review by humitos — COMMENTED** [review]
You need to define the version of the config file you want to use. It defaults to v1 where `build.jobs` is not available there.

**Comment by pamoroso** [comment]
Defining the version of the config file makes the [build of the test repo fail](https://readthedocs.org/projects/jbrtd-test/builds/17407828/) with the error `ERROR: Directory '.' is not installable. Neither 'setup.py' nor 'pyproject.toml' found`.

File: .readthedocs.yaml (hunk 1 above)
Comment (benjaoming) [reply .readthedocs.yaml:2]: Thanks @humitos :pray:

**Comment by benjaoming** [comment]
AFAICT, this error is "legitimate" in your test repo because the test doesn't have the `setup.py` that your other repo has:

image

The application has to be installed in the documentation environment because of the auto-generated reference in `reference.md`.

**Comment by pamoroso** [comment]
Adding `setup.py` to the test repo made the Read The Docs [build complete successfully](https://readthedocs.org/projects/jbrtd-test/builds/17417995/). Now the [published documentation](https://jbrtd-test.readthedocs.io/en/latest/). Has the intended content and style.

Thanks @benjaoming @humitos!

**Comment by benjaoming** [comment]
This is really great to hear! We are doing everything we can to ensure good support for Jupyter Book, happy to hear back from you and thanks for getting in touch :100: 

AFAICT, this PR should align well with the contents in your test repository and should thus be ready to merge.

**Comment by pamoroso** [comment]
@benjaoming I merged the patch to the main Suite8080 project and Read The Docs [built the documentation](https://suite8080.readthedocs.io/) [successfully](https://readthedocs.org/projects/suite8080/builds/17419023/). Your solution has the added benefit of not requiring manually editing `conf.py` to adjust the path and let `sphinx.ext.autodoc` find the Python source.

Thanks again!

**Pull request merged**